crates/loader/src/local.rs

```diff
@@ -416,7 +416,10 @@ impl LocalLoader {
                 continue;
             }
 
-            let app_root_path = src.strip_prefix(&self.app_root)?;
+            let Ok(app_root_path) = src.strip_prefix(&self.app_root) else {
+                bail!("{pattern} cannot be mapped because it is outside the application directory. Files must be within the application directory.");
```

Comment:
Oh, I just realised - preferred style is to use `quoted_path` (from the common crate) around paths destined for the UI.  This delimits the path from the surrounding text.

Reply:
for file path
> files = [{ source = "/home/ro", destination = "/catpics" }]

spin up gives the following error

> Error: Failed to load manifest from "spin.toml"
> Caused by:
>   0: Failed to load Spin app from "spin.toml"
>   1: Failed to load component `code-pls`
>   2: /home/ro/**/* cannot be mapped because it is outside the application directory. Files must be within the application directory.

I think the file path in error message gives reasonable information for the users to figure out where it went wrong :D